ideal_device/io/tlb_window.h

```diff
@@ -0,0 +1,67 @@
+/*
+ * SPDX-FileCopyrightText: (c) 2023 Tenstorrent Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#pragma once
+
+namespace tt::umd {
+
+class TLBWindow {
```

Comment:
This class seems like is redundant with the current interface.
It could be reduced to struct tlb_info {void ptr, int size}.

read and write implementation can be part of core_io.h

and write8/write16 read8/read16 and others (not _block functions) can be implemented in abstract_io directly (so that the specializations actually just implement write_block and read_block